### PR TITLE
Add custom error codes

### PR DESCRIPTION
**Pull request by hewanadun**
Update CustomErrorHandler class to support custom error codes for NACC specific errors
Use the defined custom error codes in NACCValidator for error reporting
In the QC gear, these error information is used to map with NACC QC checks after validation

### CODE REVIEW

File: validator/errors.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+from cerberus.errors import (BasicErrorHandler, ErrorDefinition, ErrorTree,
+                             ValidationError)
+
+
+# pylint: disable=(too-few-public-methods)
+class ErrorDefs:
+    """ Class to define custom errors. """
+    # IMPORTANT - When adding a new error DON'T change the existing codes
+    # Error codes are used to map b/w cerberus erros and NACC error codes in QC gear
```
Comment (bjkeller) [inline validator/errors.py:13]: typo: "erros"

File: validator/nacc_validator.py [hunk 2]
```diff
@@ -143,49 +81,84 @@ def __populate_data_types(self) -> dict[str, str] | None:
...
 
         Args:
             pk_field (str): Primary key field of the project
         """
 
-        self.__pk_field: str = pk_field
+        self.__pk_field = pk_field
+
+    @property
+    def sys_erros(self) -> Dict[str, List[str]]:
```
Comment (bjkeller) [inline validator/nacc_validator.py:115]: is this a typo? sys_erros?

File: validator/errors.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    def __set_custom_error_codes(self):
+        """Add custom error codes specific to NACC validation rules
+        """
+        # IMPORTANT - When adding a new error DON'T change the existing codes
+        # Cerberus uses bit 5 and bit 7 to mark specific error types
+        # Check https://docs.python-cerberus.org/customize.html for more info
+        # Error messages are synced with ErrorDefs using the error code
+        # Error codes are also used to map b/w cerberus erros and NACC error codes in QC gear
+        custom_errors = {
```
Comment (bjkeller) [inline validator/errors.py:60]: Is this dict going to grow as new rules are added? And, if so is there a way to handle them dynamically

**Review by bjkeller — APPROVED** [review]
There are some occurrences of the string "erros" that appear to be typos.

And, I just had a question about whether the error code map is going to have to be maintained as the rules change. But that doesn't require a fix.

Looks good.

File: validator/errors.py (hunk 1 above)
Comment (hewanadun) [reply validator/errors.py]: Fixed

File: validator/nacc_validator.py (hunk 2 above)
Comment (hewanadun) [reply validator/nacc_validator.py]: Fixed

File: validator/errors.py (hunk 3 above)
Comment (hewanadun) [reply validator/errors.py:60]: Any new rule definition using the existing rule types can be added without modifying the code. If a new rule type needs to be added, validator code has to be updated to support the new rule type, and the errors dict will also be extended alongside.

**Pull request merged**